flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/com/ververica/cdc/connectors/values/ValuesDatabase.java

```diff
@@ -0,0 +1,381 @@
+            globalTables.keySet().forEach((tableId) -> namespaces.add(tableId.getNamespace()));
+            return new ArrayList<>(namespaces);
+        }
+
+        @Override
+        public List<String> listSchemas(@Nullable String namespace)
+                throws UnsupportedOperationException {
+            Set<String> schemas = new HashSet<>();
+            globalTables.keySet().stream()
+                    .filter(
+                            (tableId) -> {
+                                if (namespace == null) {
+                                    return tableId.getNamespace() == null;
+                                } else {
+                                    return namespace.equals(tableId.getNamespace());
+                                }
+                            })
+                    .forEach(tableId -> schemas.add(tableId.getSchemaName()));
+            return new ArrayList<>(schemas);
+        }
+
+        @Override
+        public List<TableId> listTables(@Nullable String namespace, @Nullable String schemaName) {
+            return globalTables.keySet().stream()
+                    .filter(
+                            (tableId) -> {
+                                if (namespace == null) {
+                                    if (schemaName == null) {
+                                        return tableId.getNamespace() == null
+                                                && tableId.getSchemaName() == null;
+                                    } else {
+                                        return tableId.getNamespace() == null
+                                                && schemaName.equals(tableId.getSchemaName());
+                                    }
+                                } else {
+                                    // schemaName will not be null
+                                    return namespace.equals(tableId.getNamespace())
+                                            && schemaName.equals(tableId.getSchemaName());
+                                }
+                            })
+                    .collect(Collectors.toList());
+        }
+
+        @Override
+        public Schema getTableSchema(TableId tableId) {
+            return ValuesDatabase.getTableSchema(tableId);
+        }
+    }
+
+    public static Schema getTableSchema(TableId tableId) {
+        ValuesTable table = globalTables.get(tableId);
+        Preconditions.checkNotNull(table, tableId + " is not existed");
+        Schema.Builder builder = Schema.newBuilder();
+        for (Column column : table.columns) {
+            builder.physicalColumn(column.getName(), column.getType());
+        }
+        return builder.primaryKey(table.primaryKeys).build();
+    }
+
+    public static void applyDataChangeEvent(DataChangeEvent event) {
+        ValuesTable table = globalTables.get(event.tableId());
+        Preconditions.checkNotNull(table, event.tableId() + " is not existed");
```

Comment:
Will there be a problem if there is only Sink here? For example, this table was created before and is not in globalTables. At this time, the event will not be synchronized.

Reply:
DataSource is required to emit CreateTableEvent once before passing all DataChangeEvents.

Reply:
Does it mean that CreateTableEvent will be sent every time before sending DataChangeEvent?

Reply:
Yeah, DataSource will send the CreateTableEvent before sending a DataChangeEvent of a new tableId.

Reply:
We can add note to java docs of `CreateTableEvent` to tell uses that sink should use `create if not exists` statement when apply a `CreateTableEvent`